optaplanner-examples/src/main/java/org/optaplanner/examples/machinereassignment/solver/score/MachineReassignmentConstraintProvider.java

```diff
@@ -53,31 +62,28 @@
      * Maximum capacity: The maximum capacity for each resource for each machine must not be exceeded.
      */
     private Constraint maximumCapacity(ConstraintFactory factory) {
-        throw new UnsupportedOperationException("Not yet implemented due to missing support for bi-grouping.");
-//        return factory.from(MrMachineCapacity.class)
-//                .join(MrProcessAssignment.class,
-//                        equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine)
-//                )
-//                .groupBy(
-//                        (machineCapacity, processAssignment) -> machineCapacity, sumLong(
-//                                (machineCapacity, processAssignment) -> processAssignment.getUsage(machineCapacity.getResource())
-//                        )
-//                )
-//                .filter(((machineCapacity, usage) -> machineCapacity.getMaximumCapacity() < usage))
-//                .penalizeLong(MrConstraints.MAXIMUM_CAPACITY,
-//                        HardSoftLongScore.ofHard(1L),
-//                        (machineCapacity, usage) -> machineCapacity.getMaximumCapacity() - usage);
+        return factory.from(MrMachineCapacity.class)
+                .join(MrProcessAssignment.class,
+                      equal(MrMachineCapacity::getMachine, MrProcessAssignment::getMachine)
+                )
+                .groupBy((machineCapacity, processAssignment) -> machineCapacity, sumLong(
+                        (machineCapacity, processAssignment) -> processAssignment.getUsage(machineCapacity.getResource()))
+                )
```

Comment:
One experiment to make this go faster (not to be done as part of this PR!),
is to something like this:

``
from(MrProcessAssignment) // Look mom, no resource selected!
.groupBy(MrProcessAssignment::getMachine,
    sum(MrProcessAssignment::getUsages, some lamdba's)) // MrProcessAssignment::getUsages returns an array of usages
```

Now that would screw up our constraint matches count, etc but the score would end up the same. Regardless if we'd want to do it, it's an interesting performance experiment.